docs(widget): sounds, tokens, images, banners

shoom3301 opened the pull request:
# Description

Docs for the features:

- [feat(widget): use custom sounds in widget](https://github.com/cowprotocol/cowswap/pull/4162)
- [feat(widget): use custom tokens as virtual list](https://github.com/cowprotocol/cowswap/pull/4164)
- [feat(widget): customize images](https://github.com/cowprotocol/cowswap/pull/4159)
- [feat(widget): option to hide safe banner](https://github.com/cowprotocol/cowswap/pull/4175)

shoom3301 committed:
docs(widget): sounds, tokens, images, banners

elena-zh commented:
Hey @shoom3301 , great!
However, some issues/questions:
1. I noticed that not only MP3 sounds can be supported, but WAV and OGG files as well. WDYT about adding this info as well? 
![ogg and wav](https://github.com/cowprotocol/docs/assets/70885163/4c7bd9df-4d81-4d5b-a267-09fec260c036)


2. I'd add here qoutes at least. Or maybe formatted 'null' as a code
![as  a code](https://github.com/cowprotocol/docs/assets/70885163/65732931-62e7-4b9a-99c2-d60bd0d12b77)
3. For images, WDYT about adding supported formats and sizes? :) 
![fromats](https://github.com/cowprotocol/docs/assets/70885163/ebb9ba1d-48aa-464e-a195-f3da7e5fa710)
If not in the table, but maybe inside the 'Customization' section

Thank you!

anxolin commented on docs/cow-protocol/tutorials/widget/widget.md:
```diff
@@ -131,24 +131,28 @@ createCowSwapWidget(
+| `customTokens`         | `TokenInfo[]`          | ---                    | In the case when you want to use tokens that are not included in the default token lists, you can add them programmatically by simply passing an array of ERC-20 tokens info.                                                                                                                                               |
+| `disableToastMessages` | `boolean`              | false                  | CoW Swap displays a pop-up notification when certain events occur, for example: an order has been filled. You may want to handle these events yourself and disable the display of notifications in the widget, to do this you need to enable this option. See [Events handling](#events-handling) section for more details. |
+| `standaloneMode`       | `boolean`              | false                  | When this option is enabled, the widget will use its own Ethereum provider and the user will be able to connect a wallet from within the widget.                                                                                                                                                                            |
+| `hideLogo`             | `boolean`              | false                  | Option to hide the logo in the widget.                                                                                                                                                                                                                                                                                      |
```
do we still use this setting? do we need it?

shoom3301 replied:
Nice catch!
Removed it

anxolin commented on docs/cow-protocol/tutorials/widget/widget.md:
```diff
@@ -131,24 +131,28 @@ createCowSwapWidget(
+| `images`               | `CowSwapWidgetImages`  | ---                    | To make your widget unique, you can replace some of the images. For example, a stub image when there are no limit orders.                                                                                                                                                                                                   |
+| `banners`              | `CowSwapWidgetBanners` | ---                    | With this option you can control the display of some banners in the CoW Swap interface.                                                                                                                                                                                                                                     |
+| `tokenLists`           | `string[]`             | ---                    | CowSwap uses the [token lists standard](https://tokenlists.org), which allows flexible and decentralized management of assets for trading.                                                                                                                                                                                  |
+| `customTokens`         | `TokenInfo[]`          | ---                    | In the case when you want to use tokens that are not included in the default token lists, you can add them programmatically by simply passing an array of ERC-20 tokens info.                                                                                                                                               |
```
Probably nice to mention also `tokenLists` option as its related

shoom3301 replied:
Fixed

anxolin commented on docs/cow-protocol/tutorials/widget/widget.md:
```diff
@@ -131,24 +131,28 @@ createCowSwapWidget(
+| `sounds`               | `CowSwapWidgetSounds`  | ---                    | The famous MOOO! sound can be changed to something else, just provide a link to the media file (mp3). You can also specify a value of null to disable the sound.                                                                                                                                                            |
+| `images`               | `CowSwapWidgetImages`  | ---                    | To make your widget unique, you can replace some of the images. For example, a stub image when there are no limit orders.                                                                                                                                                                                                   |
+| `banners`              | `CowSwapWidgetBanners` | ---                    | With this option you can control the display of some banners in the CoW Swap interface.                                                                                                                                                                                                                                     |
+| `tokenLists`           | `string[]`             | ---                    | CowSwap uses the [token lists standard](https://tokenlists.org), which allows flexible and decentralized management of assets for trading.                                                                                                                                                                                  |
```
We could maybe add `See also [customTokens] option`  as they are related

shoom3301 replied:
Fixed

anxolin commented on docs/cow-protocol/tutorials/widget/widget.md:
```diff
@@ -131,24 +131,28 @@ createCowSwapWidget(
+| `env`                  | `CowSwapWidgetEnv`     | 'prod'                 | The environment of the widget (`local` , `prod` , `dev` , `pr`). See [`COWSWAP_URLS`](https://github.com/cowprotocol/cowswap/blob/develop/libs/widget-lib/src/consts.ts) const value for urls.                                                                                                                              |
+| `tradeAssets`          | `TradeAssets`          | Same as in swap.cow.fi | An object containing information about the selling and buying assets. Example: `{ asset: 'WBTC', amount: 12 }` or `{ asset: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48' }`.                                                                                                                                                |
+| `theme`                | `CowSwapTheme`         | 'light'                | The theme of the widget (`'dark'` for dark theme or `'light'` for light theme). It is also possible to set your own colors (`CowSwapWidgetPalette`). See [Custom theme](#custom-theme) section for more details.                                                                                                            |
+| `sounds`               | `CowSwapWidgetSounds`  | ---                    | The famous MOOO! sound can be changed to something else, just provide a link to the media file (mp3). You can also specify a value of null to disable the sound.                                                                                                                                                            |
```
❤️

alfetopito commented on docs/cow-protocol/tutorials/widget/widget.md:
```diff
@@ -187,6 +191,72 @@ Try it yourself: https://widget.cow.fi.
+const container = document.getElementById('cowswap-widget')
+
+const params: CowSwapWidgetParams = {
+    tokenLists: ['https://gateway.ipfs.io/ipns/tokens.uniswap.org']
```
Can we suggest by default a token list that's our own instead of uniswap's?

shoom3301 replied:
Set https://files.cow.fi/tokens/CowSwap.json

shoom3301 committed:
chore: fix docs

shoom3301 commented:
@elena-zh thank you!

1. Added "mp3 or any other audio format supported by your browser"
2. Fixed
3. I think this is not necessary, integrators should know best which formats and sizes they want to use

shoom3301 committed:
chore: fix docs

shoom3301 committed:
chore: fix docs
